src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_scope_NES.c

```diff
@@ -277,10 +311,91 @@ void KaleidoScope_DrawTexQuadRGBA32(GraphicsContext* gfxCtx, TexturePtr texture,
     CLOSE_DISPS(gfxCtx);
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_kaleido_scope/func_80821CC4.s")
+#define SWITCH_PAGE_LEFT 0
+#define SWITCH_PAGE_RIGHT 2
 
-void KaleidoScope_HandlePageToggles(PlayState* play, Input* input);
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_kaleido_scope/KaleidoScope_HandlePageToggles.s")
+void KaleidoScope_SwitchPage(PauseContext* pauseCtx, u8 direction) {
+    pauseCtx->cursorSlot[pauseCtx->pageIndex] = 0;
+    pauseCtx->mainState = PAUSE_MAIN_STATE_SWITCHING_PAGE;
+    pauseCtx->switchPageTimer = 0;
+
+    if (direction == SWITCH_PAGE_LEFT) {
+        pauseCtx->nextPageMode = (pauseCtx->pageIndex * 2) + 1;
+        play_sound(NA_SE_SY_WIN_SCROLL_LEFT);
+        pauseCtx->cursorSpecialPos = PAUSE_CURSOR_PAGE_RIGHT;
+    } else { // SWITCH_PAGE_RIGHT
+        pauseCtx->nextPageMode = pauseCtx->pageIndex * 2;
+        play_sound(NA_SE_SY_WIN_SCROLL_RIGHT);
+        pauseCtx->cursorSpecialPos = PAUSE_CURSOR_PAGE_LEFT;
+    }
+
+    gSaveContext.buttonStatus[EQUIP_SLOT_B] = BTN_ENABLED;
+    gSaveContext.buttonStatus[EQUIP_SLOT_C_LEFT] = BTN_DISABLED;
+    gSaveContext.buttonStatus[EQUIP_SLOT_C_DOWN] = BTN_DISABLED;
+    gSaveContext.buttonStatus[EQUIP_SLOT_C_RIGHT] = BTN_DISABLED;
+    gSaveContext.buttonStatus[EQUIP_SLOT_A] = BTN_DISABLED;
+
+    gSaveContext.hudVisibility = HUD_VISIBILITY_IDLE;
+    Interface_SetHudVisibility(HUD_VISIBILITY_ALL);
+}
+
+/**
+ * Process inputs to decide whether to switch pages
+ */
+void KaleidoScope_HandlePageToggles(PlayState* play, Input* input) {
+    PauseContext* pauseCtx = &play->pauseCtx;
+    InterfaceContext* interfaceCtx = &play->interfaceCtx;
+
+    //! FAKE
+    if (1) {}
+
+    if ((pauseCtx->debugEditor != DEBUG_EDITOR_NONE) || pauseCtx->itemDescriptionOn) {
+        return;
+    }
+
+    if (CHECK_BTN_ALL(input->cur.button, BTN_DRIGHT) || CHECK_BTN_ALL(input->press.button, BTN_R)) {
+        // Switch the page to the right regardless of where the cursor is
+        if (interfaceCtx->unk_212 == 6) {
+            func_8011552C(play, 0x15);
+        }
+        KaleidoScope_SwitchPage(pauseCtx, SWITCH_PAGE_RIGHT);
+        return;
+    }
+
+    if (CHECK_BTN_ALL(input->cur.button, BTN_DLEFT) || CHECK_BTN_ALL(input->press.button, BTN_Z)) {
+        // Switch the page to the left regardless of where the cursor is
+        if (interfaceCtx->unk_212 == 6) {
+            func_8011552C(play, 0x15);
+        }
+        KaleidoScope_SwitchPage(pauseCtx, SWITCH_PAGE_LEFT);
+        return;
+    }
+
+    if (pauseCtx->cursorSpecialPos == PAUSE_CURSOR_PAGE_LEFT) {
```

Comment:
else if definitely matches, and is more to our style, but I find it slightly easier to read with the returns since it breaks it up a little. But can change to `else if` if we want

Reply:
It's fine either way. Early returns can help breaking the logic a bit into more sensible chunks, so it is fine